### PR TITLE
add docs: Deploy a simplest multi-cluster Deployment

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Add docs about how to `Deploy a simplest multi-cluster Deployment`.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
Hi @Tingtal, can you help review it?

File: docs/userguide/resource-propagating.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+```shell
+The karmadactl get command now only supports Push mode, [ member3 ] are not push mode
+
+NAME                     CLUSTER   READY   STATUS    RESTARTS   AGE
+nginx-6799fc88d8-s7vv9   member1   1/1     Running   0          52s
+```
+
+### Update PropagationPolicy
+
+You can update the propagationPolicy by applying a new YAML file. This YAML file propagate the deployment to member2 clusters.
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:71]:
```suggestion
You can update the propagationPolicy by applying a new YAML file. This YAML file propagates the Deployment to the member2 cluster.
```

File: docs/userguide/resource-propagating.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+metadata:
+  name: example-policy
+spec:
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      name: nginx
+  placement:
+    clusterAffinity:
+      clusterNames: # Modify the selected cluster to be propagated.
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:86]:
```suggestion
      clusterNames: # Modify the selected cluster to propagate the Deployment.
```

File: docs/userguide/resource-propagating.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        - member2
+```
+
+1. Apply the new YAML file:
+```shell
+kubectl apply -f propagationpolicy-update.yaml
+```
+2. Display information of the deployment, the output is similar to this:
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:94]:
```suggestion
2. Display information of the Deployment (output similar to this):
```

File: docs/userguide/resource-propagating.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+```shell
+The karmadactl get command now only supports Push mode, [ member3 ] are not push mode
+
+NAME    CLUSTER   READY   UP-TO-DATE   AVAILABLE   AGE   ADOPTION
+nginx   member2   1/1     1            1           5s    Y
+```
+3. List the pods of the deployment, the output is similar to this:
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:101]:
```suggestion
3. List the pods of the Deployment (output similar to this):
```

File: docs/userguide/resource-propagating.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+nginx-6799fc88d8-8t8cc   member2   1/1     Running   0          17s
+```
+> Note: Updating the `.spec.resourceSelectors` field to change hit resources is currently not supported.
+
+### Update Deployment
+
+You can update the deployment template. The changes will be automatically synchronized to the member clusters.
+
+1. Update deployment replicas to 2
+2. Display information of the deployment, the output is similar to this:
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:115]:
```suggestion
2. Display information of the deployment (output similar to this):
```

File: docs/userguide/resource-propagating.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+```shell
+The karmadactl get command now only supports Push mode, [ member3 ] are not push mode
+
+NAME    CLUSTER   READY   UP-TO-DATE   AVAILABLE   AGE     ADOPTION
+nginx   member2   2/2     2            2           7m59s   Y
+```
+3. List the pods of the deployment, the output is similar to this:
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:122]:
```suggestion
3. List the pods of the deployment (output similar to this):
```

File: docs/userguide/resource-propagating.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+nginx-6799fc88d8-zpl4j   member2   1/1     Running   0          17s
+```
+
+### Delete a propagationPolicy
+
+Delete the propagationPolicy by name:
+```shell
+kubectl delete propagationpolicy example-policy
+```
+Delete propagationPolicy does not delete deployments that propagated to member clusters. Need to delete deployment in karmada control-plane:
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:137]:
```suggestion
Deleting a propagationPolicy does not delete Deployments propagated to member clusters. You need to delete Deployments in the karmada control-plane:
```

File: docs/userguide/resource-propagating.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+kubectl delete deployment nginx
+```
+
+## Deploy deployment into a specified set of target clusters
+
+
+## Configuring Multi-Cluster HA for Deployment
+
+
+## Multi-Cluster FailOver
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:148]:
```suggestion
## Multi-Cluster Failover
```

**Review by Tingtal — CHANGES_REQUESTED** [review]

**Comment by XiShanYongYe-Chang** [comment]
@Tingtal , thanks for your review.

**Review by RainbowMango — COMMENTED** [review]
/lgtm
/approve

File: docs/userguide/resource-propagating.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+1. Create a propagationPolicy base on the YAML file:
+```shell
+kubectl apply -f propagationpolicy.yaml
+```
+2. Create a Deployment nginx resource:
+```shell
+kubectl create deployment nginx --image nginx
+```
+> Note: The resource exists only as a template in karmada. After being propagated to a member cluster, the behavior of the resource is the same as that of a single kubernetes cluster.
```
Comment (snowplayfire) [inline docs/userguide/resource-propagating.md:43]: How to realize "The resource exists only as a template in karmada" ? stop kube-controller-manager in fed member?

Comment (snowplayfire) [reply docs/userguide/resource-propagating.md:43]: @XiShanYongYe-Chang @Tingtal

Comment (XiShanYongYe-Chang) [reply docs/userguide/resource-propagating.md:43]: We only enable part of controllers with `Kubernetes Controllers`, you can refer to [here](https://github.com/karmada-io/karmada/blob/master/docs/userguide/configure-controllers.md#kubernetes-controllers).

So when we create a resource in `karmada-apiserver`, take Deployment for example, ReplicaSet and Pod will not be generated.

Comment (snowplayfire) [reply docs/userguide/resource-propagating.md:43]: But in practice, some infra services in fed cluster are deployed as deployments and pods are needed.

Comment (XiShanYongYe-Chang) [reply docs/userguide/resource-propagating.md:43]: `karmada-controller-manager` is deployed in the host cluster, not `karmada-apiserver`.

Comment (snowplayfire) [reply docs/userguide/resource-propagating.md:43]: kube-apiserver and kube-etcd in host cluster, karmada-apiserver and karmada-etcd in fed cluster.   Is that so?

Comment (XiShanYongYe-Chang) [reply docs/userguide/resource-propagating.md:43]: `karmada-apiserver` and `karmada-etcd` are also deployed in the host cluster with namespace `karmada-system`. We can get them all with the `karmada-host` context with the command: `kubectl --kubeconfig /root/.kube/karmada.config --context  karmada-host get po -A`.

For resources that created in the `karmada-apiserver`, which are only resource templates. We can get them by `karmada-apiserver` context.

**Pull request merged**